x-pack/plugins/actions/server/builtin_action_types/xmatters.ts

```diff
@@ -0,0 +1,244 @@
+      configurationUtilities.ensureUriAllowed(secretsUrl);
+    }
+  } catch (allowListError) {
+    return i18n.translate('xpack.actions.builtin.xmatters.xmattersConfigurationError', {
+      defaultMessage: 'Error configuring xMatters action: {message}',
+      values: {
+        message: allowListError.message,
+      },
+    });
+  }
+}
+
+// action executor
+export async function executor(
+  {
+    logger,
+    configurationUtilities,
+  }: { logger: Logger; configurationUtilities: ActionsConfigurationUtilities },
+  execOptions: XmattersActionTypeExecutorOptions
+): Promise<ActionTypeExecutorResult<unknown>> {
+  const actionId = execOptions.actionId;
+  const { urlConfig, usesBasic } = execOptions.config;
+  const data = getPayloadForRequest(execOptions.params);
+
+  const secrets: ActionTypeSecretsType = execOptions.secrets;
+  const basicAuth =
+    usesBasic && isString(secrets.user) && isString(secrets.password)
+      ? { auth: { username: secrets.user, password: secrets.password } }
+      : undefined;
+  const url = usesBasic ? urlConfig : secrets.urlSecrets;
+
+  let result;
+  try {
+    if (!url) {
+      throw new Error('Error: no url provided');
+    }
+    result = await postXmatters({ url, data, basicAuth }, logger, configurationUtilities);
+  } catch (err) {
+    const message = i18n.translate('xpack.actions.builtin.xmatters.postingErrorMessage', {
+      defaultMessage: 'Error triggering xMatters workflow',
+    });
+    logger.warn(`Error thrown triggering xMatters workflow: ${err.message}`);
+    return {
+      status: 'error',
+      actionId,
+      message,
+      serviceMessage: err.message,
+    };
+  }
+
+  if (result.status >= 200 && result.status < 300) {
+    const { status, statusText } = result;
+    logger.debug(`response from xMatters action "${actionId}": [HTTP ${status}] ${statusText}`);
+
+    return successResult(actionId, data);
+  }
+
+  const message = i18n.translate('xpack.actions.builtin.xmatters.postingRetryErrorMessage', {
+    defaultMessage: 'Error triggering xMatters flow: http status {status}, retry later',
+    values: {
+      status: result.status,
+    },
+  });
+
+  return {
+    status: 'error',
+    actionId,
+    message,
+    retry: true,
+  };
+}
+
+// Action Executor Result w/ internationalisation
+function successResult(actionId: string, data: unknown): ActionTypeExecutorResult<unknown> {
+  return { status: 'ok', data, actionId };
+}
+
+interface XmattersPayload {
+  alertActionGroupName?: string;
+  signalId?: string;
+  ruleName?: string;
+  date?: string;
+  severity: string;
+  spaceId?: string;
+  tags?: string;
+}
+
+function getPayloadForRequest(params: ActionParamsType): XmattersPayload {
+  const data: XmattersPayload = {
+    alertActionGroupName: params.alertActionGroupName,
+    signalId: params.signalId,
+    ruleName: params.ruleName,
+    date: params.date,
+    severity: params.severity || 'High',
+    spaceId: params.spaceId,
+    tags: params.tags,
```

Comment:
Since these are set to defaults in the client code, if a user uses the API to trigger a workflow, none of these values will be set. Should we set defaults here as well if the params are undefined?

cc @pmuellr

Reply:
Do you mean like setting default strings? xMatters will set anything that doesn't have the signalId and the alertActionGroupName as a test signal.

Reply:
Probably worth noting that as a comment then.  Otherwise we'll be wondering about this in the future! :-)